### ISSUE TITLE
TextInput Bug - React Native ^0.54

### ISSUE THREAD

**Issue by myrsapp** [facebook/react-native#18844]
Environment:
OS: macOS Sierra 10.12.6
Node: 8.9.4
Yarn: 0.24.6
npm: 5.6.0
Watchman: 4.9.0
Xcode: Xcode 9.3 Build version 9E145

Packages: (wanted => installed)
 react: ^16.3.0-alpha.1
 react-native: ^0.54.4

I am using code example from https://facebook.github.io/react-native/docs/textinput.html, but I change value to "HARD CODE". If I try on the website, I cannot change the value. However, if i make one on my project, i can change the value.

```
import React, { Component } from 'react';
import { AppRegistry, TextInput } from 'react-native';

export default class UselessTextInput extends Component {
  constructor(props) {
    super(props);
    this.state = { text: 'Useless Placeholder' };
  }

  render() {
    return (
      <TextInput
        style={{height: 40, borderColor: 'gray', borderWidth: 1}}
        onChangeText={(text) => this.setState({text})}
        value={"HARD CODE"}
      />
    );
  }
}
```

**Comment by Jerameel** [comment]
Also facing the same issue on iOS. I had a validation with textInput and it changes the value state, but on iOS the value property is being overriden by the original(should be the validated value from state) value. It works fine on Android though.

**Comment by chefkefu** [comment]
This doesn't seem to be a problem in version 0.51 and older but got regressed in the newer version of react native.

<TextInput onChangeText={(text) => this.setState({text: "Hello"})} value={this.state.text} ></TextInput>

**Comment by benhurott** [comment]
Same here.

**Comment by zaguini** [comment]
Any fixes?

**Comment by Fconstant** [comment]
Same. This also affects [react-native-masked-text](https://github.com/benhurott/react-native-masked-text)

**Comment by GregorHorvatH** [comment]
Same Problem.
react-native: **0.55.4**

**Comment by vovkasm** [comment]
#18874

**Comment by jackmew** [comment]
No Problem:
0.51.0 React 16.0.0 (Original for my project)
0.53.0 React 16.2.0

Cause Problem:
0.54.0 [React 16.3.0-alpha.1(Context API)](https://reactjs.org/blog/2018/03/29/react-v-16-3.html%20), [A lot of under-the-covers work on Yoga, iOS's Text and TextInput](https://github.com/facebook/react-native/releases/tag/v0.54.0)
0.55.0
0.55.2
0.55.3
0.55.4

**Comment by garfieldnate** [comment]
Probably being solved in this PR: https://github.com/facebook/react-native/pull/18278. There are many separate bug reports for the same bug in TextInput.

**Comment by Balasnest** [comment]
Faced same problem in **0.55.4** What is the **stable version** for this fix? Should we go for **0.56** ?

**Comment by liamse** [comment]
I got the same problem in **0.56.0** but in **0.57.0** it is solved.

**Comment by kelset** [comment]
Given the inactivity and that latest comments say that is fixed on 0.57 I'm going to close this.

Please open a new issue following the template if needed.
